db/changes.go

```diff
@@ -282,6 +292,20 @@ func (db *Database) buildRevokedFeed(singleChannelCache SingleChannelCache, opti
 	return feed
 }
 
+func UserStillMaintainsAccessToDoc(db *Database, docID, revID string) (bool, error) {
+	rev, err := db.revisionCache.Get(docID, revID, false, false)
+	if err != nil {
```

Comment:
On a KeyNotFound error, I think we can return "false, nil", as they don't have access to a rev that doesn't exist. Otherwise return error.